pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -253,8 +253,11 @@ func (f *framework) RunPermitPlugins(
 	if statusCode == Wait {
 		w := newWaitingPod(pod)
 		f.waitingPods.add(w)
-		defer f.waitingPods.remove(pod.UID)
 		timer := time.NewTimer(timeout)
+		defer func() {
+			f.waitingPods.remove(pod.UID)
+			timer.Stop()
```

Comment:
why do we need to stop the timer? at the time this defer function is called, the timer should be garbage collected anyway.